packages/search/types.ts

```diff
@@ -0,0 +1,8 @@
+export interface AlgoliaConfigObject {
```

Comment:
Added a types file as `react-docs-page` uses the same interface for its `algoliaConfig` prop. This seems to be a common pattern in our component library, that we're starting to adopt, but wanted to call it out as I'm not sure how "blessed" it is yet.